prediction_market_agent/agents/microchain_agent/deploy.py

```diff
@@ -71,106 +108,99 @@ def run(
-            allow_stop=True,
-            long_term_memory=long_term_memory,
-            import_actions_from_memory=self.import_actions_from_memory,
-            keys=APIKeys(),
-            functions_config=self.functions_config,
-            enable_langfuse=self.enable_langfuse,
-        )
+        self.langfuse_update_current_trace(tags=[GENERAL_AGENT_TAG, self.identifier])
 
-        goal_manager = self.build_goal_manager(agent=agent)
-        goal = goal_manager.get_goal() if goal_manager else None
+        goal = self.goal_manager.get_goal() if self.goal_manager else None
         if goal:
-            agent.prompt = goal.to_prompt()
+            self.agent.prompt = goal.to_prompt()
 
         # Save formatted system prompt
-        initial_formatted_system_prompt = agent.system_prompt
+        initial_formatted_system_prompt = self.agent.system_prompt
 
         iteration = 0
-        while not agent.do_stop and (
+
+        while not self.agent.do_stop and (
             self.max_iterations is None or iteration < self.max_iterations
         ):
-            starting_history_length = len(agent.history)
+            self.before_iteration_callback()
+
+            starting_history_length = len(self.agent.history)
             try:
                 # After the first iteration, resume=True to not re-initialize the agent.
-                agent.run(iterations=1, resume=iteration > 0)
+                self.agent.run(iterations=1, resume=iteration > 0)
             except Exception as e:
                 logger.error(f"Error while running microchain agent: {e}")
                 raise e
             finally:
                 # Save the agent's history to the long-term memory after every iteration to keep users updated.
-                save_agent_history(
-                    agent=agent,
-                    long_term_memory=long_term_memory,
-                    initial_system_prompt=initial_formatted_system_prompt,
-                    # Because the agent is running in a while cycle, always save into database only what's new, to not duplicate entries.
-                    save_last_n=len(agent.history) - starting_history_length,
+                self.save_agent_history(
+                    initial_formatted_system_prompt=initial_formatted_system_prompt,
+                    save_last_n=len(self.agent.history) - starting_history_length,
                 )
-                if agent.system_prompt != initial_formatted_system_prompt:
-                    prompt_handler.save_prompt(get_editable_prompt_from_agent(agent))
+                if self.agent.system_prompt != initial_formatted_system_prompt:
+                    self.prompt_handler.save_prompt(
+                        get_editable_prompt_from_agent(self.agent)
+                    )
+
             iteration += 1
             logger.info(f"{self.__class__.__name__} iteration {iteration} completed.")
+
+            self.after_iteration_callback()
```

Comment:
`before` and `after_iteration_callback` are not microchain functions, rather something you added, right?